lib/aarch64.rb

```diff
@@ -288,6 +288,7 @@ def adr xd, label
     end
 
     def adrp xd, label
+      label = Immediate.new(label) if label.integer?
```

Comment:
Aligning with `adr` above to ensure that the implementation can handle immediate values and labels